src/main/java/org/saltyrtc/client/signaling/Signaling.java

```diff
@@ -1084,17 +1084,33 @@ private void validateNonceCookie(SignalingChannelNonce nonce) throws ValidationE
      * @param theirCookie The cookie bytes of the peer.
      * @throws ProtocolException Thrown if repeated cookie does not match our own cookie.
      */
-    void validateRepeatedCookie(Peer peer, byte[] theirCookie) throws ProtocolException {
-        // Verify the cookie
+    void validateRepeatedCookie(Cookie ourCookie, Cookie theirCookie) throws ProtocolException {
         final Cookie repeatedCookie = new Cookie(theirCookie);
-        final Cookie ourCookie = peer.getCookiePair().getOurs();
         if (!repeatedCookie.equals(ourCookie)) {
-            this.getLogger().debug("Peer repeated cookie: " + Arrays.toString(theirCookie));
+            this.getLogger().debug("Peer repeated cookie: " + Arrays.toString(theirCookie.getBytes()));
```

Comment:
@dbrgn assuming this was an oversight? Shouldn't be a problem regarding security.